be/src/column/map_column.cpp

```diff
@@ -234,14 +235,29 @@ void MapColumn::remove_first_n_values(size_t count) {
 }
 
 uint32_t MapColumn::serialize(size_t idx, uint8_t* pos) {
+    DCHECK(!_keys->is_map());
     uint32_t offset = _offsets->get_data()[idx];
     uint32_t map_size = _offsets->get_data()[idx + 1] - offset;
 
     strings::memcpy_inlined(pos, &map_size, sizeof(map_size));
     size_t ser_size = sizeof(map_size);
+
+    // unstable sort keys, map keys must be unique
+    SmallPermutation perm(map_size);
+    {
+        for (uint32_t i = 0; i < map_size; i++) {
+            perm[i].index_in_chunk = offset + i;
+        }
+        Tie tie(map_size, 1);
+        std::pair<int, int> range{0, map_size};
+        auto st = sort_and_tie_column(false, _keys, SortDesc(true, true), perm, tie, range, false);
+        DCHECK(st.ok());
+    }
+
     for (size_t i = 0; i < map_size; ++i) {
```

Comment:
For group-by queries, if its group-by-columns of json/struct types are not pruned or output, such as
```sql
select a from t group by json_column
```
can we just  cast struct/json column into string?

Reply:
the current implement is what you say